### PR TITLE
[issue-3173 2][helm] Add 2 new properties and fix tolerations

### LINKED ISSUE #3173: [helm] grafana-agent chart has some configurability missing for my use case
- `hostNetwork` for `Deamonset`
- `envFrom` to mount all keys on a secret as environment variables. Faro API keys for each app
- multiple `Ingress`, one per `app_agent_receiver_configs` server `port`
- `structuredConfig` similar to Loki, Mimir... charts for `mode: static`

I run `grafana-agent` as a `Deamonset` with `hostNetwork` to allow Kubernetes PODs to find the traces host by using the node's IP where they are running. This is a consequence of the migration from Opentelemetry collector which was able to enrich traces with the EKS and EC2 node metadata.

By provisioning multiple RUM Faro applications I need to manage various Ingresses out of the Helm chart but this could be easily done within the chart

### PR DESCRIPTION
**Pull request by carlosjgp**
#### PR Description

Add capability to specify `hostNetwork` to allow clients to find the traces collector on `HOST_IP` and `envFrom` to simplify the job of mapping multiple keys from a secret or configmap into environment variables

#### Which issue(s) this PR fixes

Fixes #3173 

#### Notes to the Reviewer

I've noticed inconsistent indentation in some templates on the chart. Reported here https://github.com/grafana/agent/issues/3341

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: operations/helm/charts/grafana-agent/values.yaml [hunk 1]
```diff
@@ -107,6 +110,9 @@ controller:
   # -- Number of pods to deploy. Ignored when controller.type is 'daemonset'.
   replicas: 1
 
+  # -- Host networking requested for this pod. Use the host's network namespace. If this option is set, the ports that will be used must be specified. Default to false.
```
Comment (rfratto) [inline operations/helm/charts/grafana-agent/values.yaml:113]:
```suggestion
  # -- Configures Pods to use the host network. When set to true, the ports that will be used must be specified.
```

**Review by rfratto — COMMENTED** [review]
Thanks! Looks good, just a few comments around documentation and the language in the CHANGELOG to help make it more clear what the issue was.

**Comment by carlosjgp** [comment]
Suggestions merged, docs updated and branch updated with changes from `main` branch

**Comment by carlosjgp** [comment]
Golang build related error on the pipeline but this PR has no related changes... flaky?

**Pull request merged**